xml-conduit/Text/XML/Stream/Parse.hs

```diff
@@ -567,25 +562,67 @@ parseIdent =
     valid '"'  = False
     valid '\'' = False
     valid '/'  = False
+    valid ';'  = False
+    valid '#'  = False
     valid c    = not $ isXMLSpace c
 
 parseContent :: DecodeEntities
              -> Bool -- break on double quote
              -> Bool -- break on single quote
              -> Parser Content
-parseContent de breakDouble breakSingle = parseEntity <|> parseTextContent where
-  parseEntity = do
+parseContent de breakDouble breakSingle = parseReference <|> parseTextContent where
+  parseReference = do
     char' '&'
-    t <- takeWhile1 (/= ';')
+    t <- parseEntityRef <|> parseHexCharRef <|> parseDecCharRef
     char' ';'
-    return $ de t
+    return t
+  parseEntityRef = do
+    TName ma b <- parseName
+    let name = maybe "" (`T.append` ":") ma `T.append` b
```

Comment:
Given that `:` is actually allowed by the XML specification, maybe we could simplify `parseName` implementation and this line together.